src/tcgwars/logic/groovy/TcgStatics.groovy

```diff
@@ -412,9 +412,27 @@ class TcgStatics {
   static healAll(PokemonCardSet target, Source source=Source.ATTACK){
     bg().em().run(new RemoveDamageCounter(target,source,target.damage))
   }
+  //Leaving this in since legacy code probably depends on it.
   static reduceDamageNextTurn (HP reduce, Move thisMove){
     new ReduceDamageNextTurn(reduce, thisMove.name).run(bg())
   }
+  static reduceDamageNextTurn (HP reduce, Move thisMove, PokemonCardSet self){
+    afterDamage {
+      delayed {
+        before APPLY_ATTACK_DAMAGES, {
+          bg.dm().each {
+            if(it.to == self && it.dmg.value){
```

Comment:
Wouldn't `&& it.notNoEffect` be needed over here? Shred/Swift attacks should not decrease the damage inflicted when attacking.

Reply:
You are correct but for some reason in my testing switf/shred damage went through just fine without it.

Reply:
Huh, weird.